packages/client/src/components/Filters/Filters.tsx

```diff
@@ -98,10 +100,11 @@ const CheckboxList = ({ listLabel, filters, filtersName, selectedCheckboxes }: C
 };
 
 const Filters = () => {
-  const [checked, setChecked] = useState<{ name: string | null; value: string | null }[]>([]);
+  const [checked, setChecked] = useState<SingleFilterType[]>([]);
+  const { addFilters } = useFiltersContext();
 
-  const handleChecked = (e: ChangeEvent<HTMLInputElement>) => {
-    let updatedList = [...checked];
+  const handleChecked = (e: React.ChangeEvent<HTMLInputElement> & React.MouseEvent<HTMLDivElement>): void => {
+    let updatedList: SingleFilterType[] = [...checked];
 
     if (e.target?.getAttribute('type') === 'checkbox') {
```

Comment:
Tutaj tak na szybko sprawdzenie czy `e.target` jest input-checkbox-em można zrobić tak:
```suggestion
    if (isInputCheckbox(e.target)) {
```
Przy czym `isInputChecbox` może wyglądać tak:
```ts
function isInputCheckbox(value: any): value is HTMLInputElement {
  return value !== null && value.type && value.type === 'checkbox';
}
```